### PR TITLE
use dynamic imports, remove usage of `require()`

### LINKED ISSUE #1606: GraphiQL depends on require()
I'm trying to use GraphiQL with snowpack and this highlighted that GraphiQL uses `require` internally.  Now that there's a standard for a browser implementation of dynamic loading, can this be refactored to use dynamic import instead? Having the library be dependent upon webpack or browserify feels like the wrong (or at least an out of date) solution.

### PR DESCRIPTION
**Pull request by acao**
resolves #1606 by replacing usage of legacy inline `require()` pattern with dynamic import, which is now available in all browsers we target

among other benefits, this allows tools like `esbuild` and `vite` to have a better time with this library

### CODE REVIEW

**Comment by acao** [comment]
@timsuchanek merging this one as well, as part of 1.2!

**Comment by acao** [comment]
@timsuchanek lets' merge this one before release as well!

File: packages/graphiql/src/components/HeaderEditor.tsx [hunk 1]
```diff
@@ -59,21 +61,68 @@ export class HeaderEditor extends React.Component<HeaderEditorProps> {
...
+        }}
+        ref={node => {
+          this._node = node as HTMLDivElement;
+        }}
+      />
+    );
+  }
+
+  // @ts-expect-error
+  addonModules = () => [import('codemirror/mode/javascript/javascript')];
```
Comment (timsuchanek) [inline packages/graphiql/src/components/HeaderEditor.tsx:122]: What would the TS error be here?

**Review by timsuchanek — APPROVED** [review]
Looks good to me!

File: packages/graphiql/src/components/HeaderEditor.tsx (hunk 1 above)
Comment (acao) [reply packages/graphiql/src/components/HeaderEditor.tsx:122]: that there isn't a type definition for this for `@types/codemirror`, just a legacy codemirror 5 issue

Comment (timsuchanek) [reply packages/graphiql/src/components/HeaderEditor.tsx:122]: Ok, thanks for the context! So the afterworld can find this context in this PR

Comment (acao) [reply packages/graphiql/src/components/HeaderEditor.tsx:122]: I could just properly override them in a .d.ts file I guess, but this was easier?

**Comment by acao** [comment]
thanks @timsuchanek ! now I'm going to add some esbuild/vite examples for graphql!

**Pull request merged**